processing/src/main/java/org/apache/carbondata/processing/loading/sort/unsafe/UnsafeSortDataRows.java

```diff
@@ -94,21 +94,30 @@
 
   private final long taskId;
 
-  public UnsafeSortDataRows(SortParameters parameters,
-      UnsafeIntermediateMerger unsafeInMemoryIntermediateFileMerger, int inMemoryChunkSize) {
-    this.parameters = parameters;
-    this.tableFieldStat = new TableFieldStat(parameters);
-    this.rowBuffer = new ThreadLocal<ByteBuffer>() {
+  public static ThreadLocal<ByteBuffer> getRowBuffer() {
+    return rowBuffer;
+  }
+
+  static  {
+    rowBuffer = new ThreadLocal<ByteBuffer>() {
```

Comment:
Please try using DataOutSteam backed by ByteOutPutStream, it can expand dynamically.